Green-Kittens/bathroom-finder-frontend
# Create Facility on Facility Form

# Checklist

- [x] Have I clearly explained the issue or feature request that this PR addresses?
- [x] Have I provided where new implementations for the PR have been made?
- [x] Did I highlight any areas that may need special attention during the review?
- [x] If applicable, did I specify any setup requirements or dependencies for testing?
- [x] Have I disclosed any known issues or limitations related to this PR? Did I explain why these limitations exist and how they might be addressed in the future?
- [x] Have I provided enough context for someone unfamiliar with the issue or the codebase to understand the changes?
- [x] Have I assigned appropriate reviewers who are knowledgeable about the changes? Did I consider including team members who might be affected or interested in the changes?

# Description

## Summary
Issue #145:
- original implementation did not pass facility form inputs to backend
- new implementation creates a facility in the db using the facility form inputs
- facilities can now be made using the form and will render as a marker on the main map

## Expectations
- Steps for Testing: 
   1. Navigate to the facility form
   2. Wait for the map view to render-- implementation will not allow for the creation of a facility if current location has not yet been fetched (markerAddress must have a value)
   3. Fill in the form
   4. Submit the facility
       - Expect to navigate back to the main screen 
       - The main map should now reload with the newly created facility 
       - if you click the facility marker, expect to navigate to its facility profile where you should see your inputs:
           - Name of Facility: address of marker 
           - Image Carousel: should populate with your uploaded images 
           - Add Review button: should navigate to review form but implementation to render facility name has not been done yet
           - Star ratings should by default be at 5 stars
           - Operation Hours, Category (should be "None"), Tags chosen, and description input should display in collapse view

## Additional Comments/Notes
** May need to reload main to see new facility-- sometimes works sometimes does not 
- Controller Function, createBathroom is now in bathroomController.tsx-- was originally in userController.tsx
- Made some edits to createBathroom to correctly pass in the required parameters for the creation of a facility
